torch/utils/benchmark/utils/timer.py

```diff
@@ -2,11 +2,17 @@
 
 import timeit
 import textwrap
-from typing import Any, Callable, Dict, List, NoReturn, Optional
+from typing import cast, Any, Callable, Dict, List, NoReturn, Optional, Type, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Protocol
+else:
+    Protocol = object
```

Comment:
Euh, is this a BC thing?

Reply:
MyPy has a fit if you do:
```
try:
    from typing import Protocol
except ImportError:
    from typing_extensions import Protocol
```

and the `Protocol = object` trick is how other parts of the code handled that. I quite like your suggestion above of using an assert rather than a cast, although it would (I think) require the rather comical:
```
if TYPE_CHECKING:
    from typing import Protocol
else:
    try:
        from typing import Protocol
    except ImportError:
        from typing_extensions import Protocol
```